tests/system/src/test/java/test/com/sun/javafx/application/InitializeJavaFXBase.java

```diff
@@ -0,0 +1,98 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.com.sun.javafx.application;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.stage.Stage;
+import junit.framework.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class InitializeJavaFXBase {
+
+    public static final CountDownLatch appLatch = new CountDownLatch(1);
+
+    public static class InitializeApp extends Application {
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            appLatch.countDown();
+        }
+    }
+
+    public static void initializeApplicationLaunch() throws Exception {
+        new Thread(() -> {
+            Application.launch(InitializeApp.class);
+        }).start();
+        appLatch.await(5, TimeUnit.SECONDS);
+    }
+
+    public static void initializeStartup() throws Exception {
+        CountDownLatch latch = new CountDownLatch(1);
+        Platform.startup(() -> {
+            latch.countDown();
+        });
+        latch.await(5, TimeUnit.SECONDS);
+    }
+
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            System.out.println("start called!");
+        }
+    }
+
+    public void doTestInitializeThenLaunchInFX() throws Exception {
+        Util.runAndWait(() ->{
+            try {
+                System.out.println("Calling launch!");
+                Application.launch(TestApp.class);
+                System.out.println("Finished launch!");
+                Assert.fail("Error: No Exception was thrown - expected IllegalStateException");
+            } catch (IllegalStateException e) {
+            }
+        });
+    }
+
+    public void doTestInitializeThenSecondLaunch() throws Exception {
+        try {
+            System.out.println("Calling launch!");
+            Application.launch(TestApp.class);
+            System.out.println("Finished launch!");
+            Assert.fail("Error: No Exception was thrown - expected IllegalStateException");
+            throw new Exception();
+        } catch (IllegalStateException e) {
+            System.out.println("Works!");
+        } catch (Exception e) {
+            throw e;
+        }
```

Comment:
Line 91 is reachable. Application.launch throws an IllegalStateException, even so, it doesn't declare it.
I've Removed the second catch block.

Reply:
At the time Ambarish made the comment, line 91 referred to this statement:

```
             throw new Exception();
```

This is indeed not reachable (although the compiler can't tell that), since `Assert.fail` unconditionally throws an `AssertionError`